frontend/src/metabase-lib/lib/queries/StructuredQuery.ts

```diff
@@ -1590,6 +1559,12 @@ class StructuredQueryInner extends AtomicQuery {
    * returns the original Table object at the beginning of the nested queries
    */
   rootTable(): Table {
+    const question = this.question();
+    const questionTableId = question?.tableId();
+    if (questionTableId != null) {
+      return this.metadata().table(questionTableId);
+    }
+
     return this.rootQuery().table();
```

Comment:
If we have access to the query's question, we should check its `table_id` first because we might not always be able to resolve to the "rootQuery" -- nested cards can be however many layers deep, and we don't want to have to fetch _all_ of the underlying cards a query is based on.